projects/centre_console/inc/button_press.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+// Receives button presses on GPIO, and raises events.
+// Requires interrupts and gpio to be initialized.
```

Comment:
Just realized that most files don't have something like this, but a small "module comment" would be useful in general. Especially since when things fail due to missing inits, the error isn't verbose at all ... that or some way to actually enforce that it was initialized.

Reply:
absolutely!

Reply:
Added comments to everything.